core/modules/field/modules/options/options.module

```diff
@@ -382,3 +386,289 @@ function options_field_widget_error($element, $error, $form, &$form_state) {
+ *   token, specify a regular expression pattern that will also be allowed as
+ *   a default value. Include pattern delimiters. Defaults to an empty string.
+ *
+ *   @code
+ *   $element['options'] = array(
+ *     '#type' => 'options',
+ *     '#key_type' => 'associative',
+ *     '#key_type_toggle' => t('Custom keys'),
+ *     '#key_type_toggled' => TRUE,
+ *   );
+ *   @endcode
+ */
+function options_element_info() {
+  $type = array();
+
+  $type['options'] = array(
+    '#input' => TRUE,
+    '#process' => array('form_options_expand'),
+    '#limit' => NULL,
+    '#optgroups' => TRUE,
+    '#multiple' => FALSE,
+    '#options' => array(),
+    '#options_readonly' => FALSE,
+    '#key_type' => 'mixed',
+    '#key_type_toggle' => NULL,
+    '#key_type_toggled' => FALSE,
+    '#default_value_allowed' => TRUE,
+    '#default_value_pattern' => '',
+    '#element_validate' => array('form_options_validate'),
+    '#theme_wrappers' => array('form_element'),
+  );
+
+  return $type;
+}
+
+/**
+ * Expand the "options" form element type.
+ *
+ * The "options" type is simply an enhanced textarea that makes it easier to
+ * create key|value pairs and put items into optgroups.
+ */
+function form_options_expand($element) {
+  module_load_include('inc', 'options', 'options.element');
+  return _form_options_expand($element);
+}
+
+/**
+ * Validate the "options" form element type.
+ */
+function form_options_validate($element, &$form_state) {
+  module_load_include('inc', 'options', 'options.element');
+  _form_options_validate($element, $form_state);
+}
+
+/**
+ * This function adjusts the value of the element from a text value to an array.
+ */
+function form_type_options_value(&$element, $edit = FALSE) {
+  module_load_include('inc', 'options', 'options.element');
+  return _form_type_options_value($element, $edit);
+}
+
+/**
+ * Create a textual representation of options from an array.
+ *
+ * @param $options
+ *   An array of options used in a select list.
+ * @param $key_type
+ *   How key/value pairs should be interpreted. Available options:
+ *   - mixed
+ *   - numeric
+ *   - associative
+ *   - custom
+ *   - none
+ */
+function form_options_to_text($options, $key_type) {
+  module_load_include('inc', 'options', 'options.element');
+  return _form_options_to_text($options, $key_type);
+}
+
+/**
+ * Create an array representation of text option values.
+ *
+ * If the Key of the option is within < >, treat as an optgroup
+ *
+ * <Group 1>
+ *   creates an optgroup with the label "Group 1"
+ *
+ * <>
+ *   Exits the current group, allowing items to be inserted at the root element.
+ */
+function form_options_from_text($text, $key_type, $flat = FALSE, &$duplicates = array()) {
+  module_load_include('inc', 'options', 'options.element');
+  return _form_options_from_text($text, $key_type, $flat, $duplicates);
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function options_form_field_ui_field_settings_form_alter(&$form, &$form_state) {
```

Comment:
This should be removeable once the form definition is updated directly in `options_field_widget_form()`.